lib/src/runner/file_utils.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// File-listing utils used by dartdevrun.
+library dev_compiler.src.runner.file_utils;
+
+import 'dart:async';
+import 'dart:io';
+
+Future<List<File>> listJsFiles(Directory dir) async {
+  var list = [];
+  await for (var file in dir.list(recursive: true, followLinks: true)) {
+    if (file is File && file.path.endsWith(".js")) list.add(file);
+  }
+  return list;
+}
+
+List<String> turnFilesToUris(List<String> args) =>
```

Comment:
what's the reason to do this? we should have logic already to handle relative path -> file URI as needed
(stringToUri in compiler.dart will do this). I've been desperately trying to tame all of the path logic that sprung up in our code base, so apologies in advance for focusing so much on this